--- spreadsheet-excel-0.3.5.1/lib/spreadsheet/workbook.rb	2007-08-23 21:02:32.000000000 +1000
+++ spreadsheet-excel-0.3.5.1/lib/spreadsheet/workbook.rb	2007-08-23 21:09:24.000000000 +1000
@@ -1,3 +1,6 @@
+require 'rubygems'
+require 'ole/file_system'
+
 class Workbook < BIFFWriter
    BOF = 11
    EOF = 4
@@ -95,14 +98,12 @@
    end
    
    def store_ole_file
-      OLEWriter.open(@file){ |ole|
-         ole.set_size(@biffsize)
-         ole.write_header
-         ole.print(@data)
-         @worksheets.each{ |sheet|
-            ole.print(sheet.data)
-         }
-      }
+      Ole::Storage.open @file, 'wb+' do |ole|
+         ole.file.open 'Book', 'w' do |f|
+            f.write @data
+            @worksheets.each { |sheet| f.write sheet.data }
+         end
+      end
    end
 
    def store_window1
